### PR TITLE
feat: add revisionHistoryLimit

### PR DESCRIPTION
**Pull request by KyriosGN0**
### Type of change
- Enhancement / new feature

### Description
adding the ability to set the revisionHistoryLimit for the operator Deployment
### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [X] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: packaging/helm-charts/helm3/strimzi-kafka-operator/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 appVersion: "0.1.0"
 description: "Strimzi: Apache Kafka running on Kubernetes"
 name: strimzi-kafka-operator
-version: 0.1.1
+version: 0.2.1
```
Comment (scholzj) [inline packaging/helm-charts/helm3/strimzi-kafka-operator/Chart.yaml:5]: Please don't change this.

**Review by scholzj — COMMENTED** [review]
Thanks for the PR. I left one nit. I also suspect that the build will fail. You have to:
* Either upgrade the YAML installation files correspondingly to this change
* Wrap the Helm chart change with some condition to not set it when not configured by the user. (see how it is done for the other fields such as the `priorityClassName` as example)

I personally would prefer the second option.

**Comment by KyriosGN0** [comment]
thanks @scholzj, i opted to just set a default value in the yaml file

**Comment by scholzj** [comment]
> thanks @scholzj, i opted to just set a default value in the yaml file

That does not fix it because it will still generate the YAML with the value.

**Comment by KyriosGN0** [comment]
> > thanks @scholzj, i opted to just set a default value in the yaml file
> 
> That does not fix it because it will still generate the YAML with the value You would need to do it as it is done for example for the priority class name: https://github.com/strimzi/strimzi-kafka-operator/blob/main/packaging/helm-charts/helm3/strimzi-kafka-operator/templates/060-Deployment-strimzi-cluster-operator.yaml#L45-L47

@scholzj fixed that, but i would like to understand where is the issue? 
if a user would create a values file without this param it would have used the default..
or am i missing something?

**Comment by KyriosGN0** [comment]
hey @scholzj @ppatierno could you help me figure out what went wrong in the CI, it complains about uncommitted changes in a dir a didn't touch?
am i missing something?

**Comment by scholzj** [comment]
> hey @scholzj @ppatierno could you help me figure out what went wrong in the CI, it complains about uncommitted changes in a dir a didn't touch? am i missing something?

Yeah, this is the error I talked about. I think the Deployment YAMl in the Helm Chart looks good now. But you have to also remove the default value from the values file to not have it rendered into the YAML.

It also looks like some older commits got pulled into your PR in some rebase, could you please try to fix it as well?

**Review by scholzj — APPROVED** [review]
LGTM. Thanks.

File: packaging/helm-charts/helm3/strimzi-kafka-operator/values.yaml [hunk 2]
```diff
@@ -186,4 +187,4 @@ labelsExclusionPattern: ""
 # Controls whether Strimzi generates network policy resources (By default true)
 generateNetworkPolicy: true
 # Override the value for Connect build timeout
-connectBuildTimeoutMs: 300000
\ No newline at end of file
+connectBuildTimeoutMs: 300000
```
[the remark this answers is not in the document]
Comment (KyriosGN0) [reply packaging/helm-charts/helm3/strimzi-kafka-operator/values.yaml:189]: should i remove this line ?

File: packaging/helm-charts/helm3/strimzi-kafka-operator/Chart.yaml [hunk 3]
```diff
@@ -24,4 +24,4 @@ maintainers:
 - name: sknot-rh
 - name: PaulRMellor
 - name: im-konge
-- name: see-quick
\ No newline at end of file
+- name: see-quick
```
[the remark this answers is not in the document]
Comment (ppatierno) [reply packaging/helm-charts/helm3/strimzi-kafka-operator/Chart.yaml:27]: well it seems you remove an extra line? You don't have to remove see-quick. The change is coming from your branch, so you did it.

File: packaging/helm-charts/helm3/strimzi-kafka-operator/values.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ppatierno) [reply packaging/helm-charts/helm3/strimzi-kafka-operator/values.yaml:189]: Nope. Ditto as above. I think you removed an extra line. Again the PR is coming from your branch so you did the change.

File: packaging/helm-charts/helm3/strimzi-kafka-operator/Chart.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (KyriosGN0) [reply packaging/helm-charts/helm3/strimzi-kafka-operator/Chart.yaml:27]: seems like vs code does this for me (when i try to remove the extra white line, if i try to remove it vscode adds it again)

**Review by ppatierno — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**